Pod/Classes/WPAssetViewController.m

```diff
@@ -129,26 +138,26 @@ - (void)showVideoAsset
         if (!strongSelf) {
             return;
         }
-        [strongSelf.activityIndicatorView stopAnimating];
-        if (error || url == nil) {
-            [strongSelf showError:error];
-            return;
-        }
-        strongSelf.videoView.videoURL = url;
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [strongSelf.activityIndicatorView stopAnimating];
+            if (error || url == nil) {
+                [strongSelf showError:error];
+                return;
+            }
+            strongSelf.videoView.videoURL = url;
+        });
     }];
 }
 
 - (void)showError:(NSError *)error {
-    dispatch_async(dispatch_get_main_queue(), ^{
-        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"Media preview failed.", @"Alert title when there is issues loading an asset to preview.")
-                                                                                  message:error.localizedDescription
-                                                                           preferredStyle:UIAlertControllerStyleAlert];
-        [alertController addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"OK", @"") style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
-            [self.navigationController popViewControllerAnimated:YES];
-        }]];
-
-        [self presentViewController:alertController animated:YES completion:nil];
-    });
+    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"Media preview failed.", @"Alert title when there is issues loading an asset to preview.")
+                                                                              message:error.localizedDescription
```

Comment:
So I get a very technical looking error when iCloud images or video fail to load: "Error fetching asset with identifier: AYLnPcMZouYoRZdpCOk6EqqaZp+1 (nonpartial failure)", which isn't great to show to the user. The `NSError` does have an `underlyingError`, though with the description "The internet connection appears to be offline". So perhaps either we could check if that exists, or just show a generic "Failed to load image / video" message.